contracts/BondTellerErc20.sol

```diff
@@ -0,0 +1,464 @@
+        uint256 maturation;   // timestamp after which bond is redeemable
+    }
+
+    /// @notice mapping of bondID to Bond object
+    mapping (uint256 => Bond) public bonds;
+
+    uint256 public numBonds;
+
+    ISOLACE public solace;
+    IxSOLACE public xsolace;
+
+    uint256 internal constant MAX_BPS = 10000; // 10k basis points (100%)
+    uint256 public daoFeeBps;
+    uint256 public stakeFeeBps;
+    address public underwritingPool;
+    address public dao;
+    IBondDepository public bondDepo;
+
+    // used to determine new price
+    uint256 public nextPrice;
+    uint128 public momentumNum;
+    uint128 public momentumDenom;
+    uint256 public halfLife;
+    uint256 public lastPriceUpdate;
+
+    uint256 public maxPayout;
+
+    /***************************************
+    INITIALIZER
+    ***************************************/
+
+    /**
+     * @notice Initializes the teller.
+     * @param name_ The name of the bond token.
+     * @param governance_ The address of the [governor](/docs/protocol/governance).
+     * @param solace_ The SOLACE token.
+     * @param xsolace_ The xSOLACE token.
+     * @param pool_ The underwriting pool.
+     * @param dao_ The DAO.
+     * @param principal_ address The ERC20 token that users deposit.
+     * @param bondDepo_ The bond depository.
+     */
+    function initialize(
+        string memory name_,
+        address governance_,
+        address solace_,
+        address xsolace_,
+        address pool_,
+        address dao_,
+        address principal_,
+        address bondDepo_
+    ) external override initializer {
+        __Governable_init(governance_);
+        require(solace_ != address(0x0), "zero address solace");
+        require(xsolace_ != address(0x0), "zero address xsolace");
+        require(pool_ != address(0x0), "zero address pool");
+        require(dao_ != address(0x0), "zero address dao");
+        require(principal_ != address(0x0), "zero address principal");
+        require(bondDepo_ != address(0x0), "zero address bond depo");
+        string memory symbol = "SBT";
+        __ERC721Enhanced_init(name_, symbol);
+        solace = ISOLACE(solace_);
+        xsolace = IxSOLACE(xsolace_);
+        solace.approve(xsolace_, type(uint256).max);
+        underwritingPool = pool_;
+        dao = dao_;
+        principal = IERC20(principal_);
+        bondDepo = IBondDepository(bondDepo_);
+    }
+
+    /***************************************
+    VIEW FUNCTIONS
+    ***************************************/
+
+    // BOND PRICE
+
+    /**
+     * @notice Calculate the current price of a bond.
+     * Assumes 1 SOLACE payout.
+     * @return price_ The price of the bond measured in `principal`.
+     */
+    function bondPrice() public view returns (uint256 price_) {
+        uint256 timeSinceLast = block.timestamp - lastPriceUpdate;
+        price_ = exponentialDecay(nextPrice, timeSinceLast);
+        if (price_ < minimumPrice) {
```

Comment:
we did not set the default value for "minimumPrice" in the contract creation phase. Is it safe?

Reply:
good catch, added initialization check